### PR TITLE
feat(medusa, admin-ui, medusa-react, medusa-js): Allow toggling of manage inventory

### PR DESCRIPTION
**Pull request by pKorsholm**
**What**
- Toggle manage inventory in the inventory management modal

**How**
- Create/update/remove inventory item based on if `manage_inventory` is set and if an inventory item already exists
- Move all stock location updates to when the modal is submitted
- Add create-inventory-item endpoint in the core

Fixes CORE-1196

### CODE REVIEW

**Comment by srindom** [comment]
Super clean work! 

One thing I have found in quick testing is this: 

When untoggling the inventory item is correctly deleted. There is, however, still stale location levels in the DB. Would suggest that we make sure these are deleted as well.

File: packages/medusa/src/api/routes/admin/inventory-items/transaction/create-inventory-item.ts [hunk 1]
```diff
@@ -0,0 +1,174 @@
+import {
```
Comment (srindom) [inline packages/medusa/src/api/routes/admin/inventory-items/transaction/create-inventory-item.ts:1]: @carlos-r-l-rodrigues - can I get you to take a look at this? :)

**Review by srindom — COMMENTED** [review]
Please add integration tests for create inventory item :)

File: packages/inventory/src/services/inventory.ts [hunk 2]
```diff
@@ -245,6 +245,20 @@ export default class InventoryService
...
+      })
+
+    await Promise.all(
+      levels.map((level) =>
+        this.inventoryLevelService_
+          .withTransaction(this.activeManager_)
+          .delete(level.id)
+      )
+    )
+
```
[the remark this answers is not in the document]
Comment (srindom) [reply packages/inventory/src/services/inventory.ts:261]: might not be worth implementing at this point though.

[the remark this answers is not in the document]
Comment (srindom) [reply packages/inventory/src/services/inventory.ts:261]: Or maybe it is given this: https://github.com/medusajs/medusa/pull/3460/files/31cabe28c57aa29efce2d4cc86218cd522078d4b#diff-0014841cb319f0d12543db3bd8bd2cf990a30c762b0c16ba0881654298ba6649R211

[the remark this answers is not in the document]
Comment (adrien2p) [reply packages/inventory/src/services/inventory.ts:261]: The methods should always be able to receive an id or ids, string | string[] if we think that we might use them for bulk operations. The event will be emitted as bulk and the code must be implemented for bulk. If there is just an id then transform as [id] as we do in the places we are refactoring at the moment. Wdyt?

File: packages/medusa/src/api/routes/admin/inventory-items/transaction/create-inventory-item.ts [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+    saveResponse: true,
+    next: {
+      action: actions.attachInventoryItem,
+      noCompensation: true,
+    },
+  },
+}
+
+const createInventoryItemStrategy = new TransactionOrchestrator(
+  "create-variant-with-inventory",
```
Comment (carlos-r-l-rodrigues) [inline packages/medusa/src/api/routes/admin/inventory-items/transaction/create-inventory-item.ts:38]:
```suggestion
  "create-inventory-item",
```

File: packages/medusa/src/api/routes/admin/inventory-items/transaction/create-inventory-item.ts [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+  input: CreateInventoryItemInput
+): Promise<DistributedTransaction> => {
+  const {
+    manager,
+    productVariantService,
+    inventoryService,
+    productVariantInventoryService,
+  } = dependencies
+
+  const inventoryServiceTx = inventoryService?.withTransaction(manager)
```
Comment (carlos-r-l-rodrigues) [inline packages/medusa/src/api/routes/admin/inventory-items/transaction/create-inventory-item.ts:74]: we have decided to do not share DB transaction with modules. So you can remove the `withTransaction` everywhere and let the TO handle the compensation.

File: packages/admin-ui/ui/src/domain/products/edit/sections/variants/table.tsx [hunk 5]
```diff
@@ -106,7 +106,7 @@ const VariantsTable = ({ variants, actions }: Props) => {
 
   const getTableRowActionables = (variant: ProductVariant) => {
     const inventoryManagementActions = []
-    if (hasInventoryService && variant.manage_inventory) {
+    if (hasInventoryService) {
       inventoryManagementActions.push({
         label: "Manage inventory", // TODO: Only add this item if variant.manageInventory is true
```
Comment (srindom) [inline packages/admin-ui/ui/src/domain/products/edit/sections/variants/table.tsx:111]:
```suggestion
        label: "Manage inventory",
```

**Review by srindom — APPROVED** [review]
LGTM!

**Pull request merged**